addons/bus/static/src/services/bus_monitoring_service.js

```diff
@@ -44,13 +47,21 @@ export class BusMonitoringService {
             case WORKER_STATE.CONNECTED: {
                 this.isReconnecting = false;
                 this.isConnectionLost = false;
+                if (this.timeout) {
+                    clearTimeout(this.timeout);
+                    this.timeout = undefined;
+                }
                 break;
             }
             case WORKER_STATE.DISCONNECTED: {
                 if (this.isReconnecting) {
-                    this.isConnectionLost = true;
                     this.isReconnecting = false;
                 }
+                if (!this.timeout) {
```

Comment:
Why not put it in the above if block? Here, the notification will be shown even if no reconnect attempt was failed?

Reply:
Putting in if above would delay the showing even further, and couple the showing with a reconnect attempt.

Reply:
Well it's supposed to be coupled to a reconnect attempt. Voluntary disconnections while rare can happen, and are expected.

Reply:
Actually, this PR does not increase the delay to show the alert at all, quite the opposite. On `odoo.com`, first reconnect attempt occurs after +/- 30 seconds. So before this PR, the alert would show after +/- 30s if the reconnect attempt was not successful. IMO, it makes sense to consider the connection as lost after waiting 40s or so. 

Here, the alert will show after 15 seconds, regardless of whether we attempted a re connection or not. To me, it's a regression.

Reply:
> Here, the alert will show after 15 seconds, regardless of whether we attempted a re connection or not. To me, it's a regression and it's also incorrect.

The fact that we attempted a reconnection doesn't mean we are connected, right? The warning makes sense until the reconnection is successful.

Reply:
The fact that *we didn't* attempted a re connection when displaying the message is wrong to me. E.g. server might go down for a few seconds, this does not mean the connection is lost. Nor does it match the specs that were meant to make this message less intrusive. Nor does it handle the case were the connection was closed on purpose.